File: src/main/java/com/jewelry/pos/service/SupplierService.java
package com.jewelry.pos.service;

import com.jewelry.pos.domain.entity.Supplier;
import com.jewelry.pos.domain.repository.SupplierRepository;
import com.jewelry.pos.web.dto.SupplierDTO;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@RequiredArgsConstructor
public class SupplierService {

    private final SupplierRepository supplierRepository;

    @Transactional
    public Supplier createSupplier(SupplierDTO dto) {
        Supplier supplier = new Supplier();
        supplier.setName(dto.name());
        supplier.setPhoneNumber(dto.phoneNumber());
        supplier.setAddress(dto.address());
        supplier.setNotes(dto.notes());
        return supplierRepository.save(supplier);
    }

    @Transactional
    public Supplier updateSupplier(String id, SupplierDTO dto) {
        Supplier supplier = supplierRepository.findById(id)
                .orElseThrow(() -> new IllegalArgumentException("Supplier not found"));
        supplier.setName(dto.name());
        supplier.setPhoneNumber(dto.phoneNumber());
        supplier.setAddress(dto.address());
        supplier.setNotes(dto.notes());
        return supplierRepository.save(supplier);
    }

    @Transactional(readOnly = true)
    public List<Supplier> getAllSuppliers() {
        return supplierRepository.findAll();
    }

    @Transactional(readOnly = true)
    public Supplier getSupplierById(String id) {
        return supplierRepository.findById(id)
                .orElseThrow(() -> new IllegalArgumentException("Supplier not found"));
    }

    @Transactional
    public void deleteSupplier(String id) {
        supplierRepository.deleteById(id);
    }
}

File: frontend/src/app/core/services/supplier.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Supplier {
  id: string;
  name: string;
  phoneNumber?: string;
  address?: string;
  notes?: string;
}

export interface SupplierRequest {
  name: string;
  phoneNumber?: string;
  address?: string;
  notes?: string;
}

@Injectable({
  providedIn: 'root'
})
export class SupplierService {
  private http = inject(HttpClient);
  private readonly API_URL = '/api/suppliers';

  getAllSuppliers(): Observable<Supplier[]> {
    return this.http.get<Supplier[]>(this.API_URL);
  }

  getSupplierById(id: string): Observable<Supplier> {
    return this.http.get<Supplier>(`${this.API_URL}/${id}`);
  }

  createSupplier(request: SupplierRequest): Observable<Supplier> {
    return this.http.post<Supplier>(this.API_URL, request);
  }

  updateSupplier(id: string, request: SupplierRequest): Observable<Supplier> {
    return this.http.put<Supplier>(`${this.API_URL}/${id}`, request);
  }

  deleteSupplier(id: string): Observable<void> {
    return this.http.delete<void>(`${this.API_URL}/${id}`);
  }
}
